src/app/products/productManagement/product-view/product-view.component.ts

```diff
@@ -75,57 +76,118 @@ export class ProductViewComponent implements OnInit {
 
   prettyEdit(product: Product) {
 
-    this.product = product;
-    this.taxonomy = this.product.taxonomy;
+    this.listType = 'Pretty';
+    this.productService.findById(productId).subscribe(
+      (p) => {
+        this.product = p;
+        this.taxonomy = this.product.taxonomy;
+      });
     this.prettyShow = false;
     this.pennyShow = false;
     this.editHeaderClass = 'card-header text-white bg-secondary';
     this.editcardShow = true;
+    this.previousListType = this.product.onSale;
   }
-  pennyEdit(product: Product) {
-
-    this.product = product;
-    this.taxonomy = this.product.taxonomy;
+  pennyEdit(productId: number) {
+    this.listType = 'Penny';
+    this.productService.findById(productId).subscribe(
+      (p) => {
+        this.product = p;
+        this.taxonomy = this.product.taxonomy;
+      });
```

Comment:
afterward, assign this.product = product and this.taxonomy = this.product.taxonomy